proposals/lip-0051.md

```diff
@@ -769,14 +770,227 @@ if the above send function does not fail:
+                    "tokenID": {
+                        "type": "object",
+                        "fieldNumber": 2,
+                        "required": ["chainID", "localID"],
+                        "properties": {
+                            "chainID": {
+                                "dataType": "uint32",
+                                "fieldNumber": 1
+                            },
+                            "localID": {
+                                "dataType": "uint32",
+                                "fieldNumber": 2
+                            }
+                        }
+                    },
+                    "availableBalance": {
+                        "dataType": "uint64",
+                        "fieldNumber": 3
+                    },
+                    "lockedBalances": {
+                        "type": "array",
+                        "fieldNumber": 4,
+                        "items": {
+                            "type": "object",
+                            "required": ["moduleID", "amount"],
+                            "properties": {
+                                "moduleID": {
+                                    "dataType": "uint32",
+                                    "fieldNumber": 1
+                                },
+                                "amount": {
+                                    "dataType": "uint64",
+                                    "fieldNumber": 2
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        },
+        "supplySubstore": {
+            "type": "array",
+            "fieldNumber": 2,
+            "items": {
+                "type": "object",
+                "required": ["localID", "totalSupply"],
+                "properties": {
+                    "localID": {
+                        "dataType": "uint32",
+                        "fieldNumber": 1
+                    },
+                    "totalSupply": {
+                        "dataType": "uint64",
+                        "fieldNumber": 1
+                    }
+                }
+            }
+        },
+        "escrowSubstore": {
+            "type": "array",
+            "fieldNumber": 3,
+            "items": {
+                "type": "object",
+                "required": ["escrowChainID", "localID", "amount"],
+                "properties": {
+                    "escrowChainID": {
+                        "dataType": "bytes",
+                        "fieldNumber": 1
+                    },
+                    "localID": {
+                        "dataType": "uint32",
+                        "fieldNumber": 2
+                    },
+                    "amount": {
+                        "dataType": "uint64",
+                        "fieldNumber": 3
+                    }
+                }
+            }
+        },
+        "availableLocalIDSubstore": {
```

Comment:
Personally, I find it odd that `availableLocalIDSubstore` is actually corresponding to the one element in that substore. Maybe with additional nesting this could be made more clear, but this should be the decision by the developers (@mitsuaki-u ).

Reply:
I added a nesting, let me know it it looks good to you @mitsuaki-u , we can always remove it.

Reply:
Can be resolved from my side.